src/main/java/org/opensearch/neuralsearch/processor/RRFProcessor.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.neuralsearch.processor;
+
+import static org.opensearch.neuralsearch.search.util.HybridSearchResultFormatUtil.isHybridQueryStartStopElement;
+
+import java.util.stream.Collectors;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+import org.opensearch.neuralsearch.processor.combination.ScoreCombinationTechnique;
+import org.opensearch.neuralsearch.processor.normalization.ScoreNormalizationTechnique;
+import org.opensearch.search.fetch.FetchSearchResult;
+import org.opensearch.search.query.QuerySearchResult;
+
+import org.opensearch.action.search.QueryPhaseResultConsumer;
+import org.opensearch.action.search.SearchPhaseContext;
+import org.opensearch.action.search.SearchPhaseName;
+import org.opensearch.action.search.SearchPhaseResults;
+import org.opensearch.search.SearchPhaseResult;
+import org.opensearch.search.internal.SearchContext;
+import org.opensearch.search.pipeline.SearchPhaseResultsProcessor;
+
+import lombok.AllArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+
+/**
+ * Processor for implementing reciprocal rank fusion technique on post
+ * query search results. Updates query results with
+ * normalized and combined scores for next phase (typically it's FETCH)
+ * by using ranks from individual subqueries to calculate 'normalized'
+ * scores before combining results from subqueries into final results
+ */
+@Log4j2
+@AllArgsConstructor
+public class RRFProcessor implements SearchPhaseResultsProcessor {
+    public static final String TYPE = "score-ranker-processor";
```

Comment:
Why is this processor named as `score-ranker-processor`? What are supported techniques other than `rrf`?

Reply:
Hi @martin-gaievski ! Can you respond to my question before resolving this conversation?